File: my-folder/0049-group-anagrams/solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
         List<List<String>> anagramList=new ArrayList<>();
       HashMap<String,List<String>> anagramMap=new HashMap<>();
        for(String str:strs)
        {
            char[] strsArr = str.toCharArray();
            Arrays.sort(strsArr);
            String strNew=new String(strsArr);
            if(anagramMap.containsKey(strNew))
            {
                List<String> strings = anagramMap.get(strNew);
                strings.add(str);
                anagramMap.put(strNew,strings);
            }
            else{
                ArrayList<String> stringArrayList = new ArrayList<>();
                stringArrayList.add(str);
                anagramMap.put(strNew, stringArrayList);
            }
        }
         anagramMap.values().stream().forEach(value->anagramList.add(value));
        return anagramList;
        
    }
}

File: my-folder/0049-group-anagrams/solution.kt
class Solution {
    fun groupAnagrams(strs: Array<String>): List<List<String>> {
        val anagramMap=mutableMapOf<String, MutableList<String>>()
        strs.forEach()
        { word ->
             val indexArray=IntArray(26)
              word.forEach()
                {
                    ch->indexArray[ch-'a']++
                }
              val key=indexArray.joinToString()
              anagramMap.getOrPut(key){mutableListOf()}.add(word)
            }
            return anagramMap.values.toList()
    }
}
